nexus_system_orchestrator/src/system_orchestrator.cpp

```diff
@@ -295,6 +292,54 @@ auto SystemOrchestrator::on_configure(const rclcpp_lifecycle::State& previous)
     return CallbackReturn::FAILURE;
   }
 
+  // create action server for bidding for transporters
+  this->_bid_transporter_srv =
+    rclcpp_action::create_server<BidTransporterActionType>(
+      this->get_node_base_interface(),
+      this->get_node_clock_interface(),
+      this->get_node_logging_interface(),
+      this->get_node_waitables_interface(),
+      BidTransporterAction::action_name(),
+      [this](const rclcpp_action::GoalUUID& uuid,
+        std::shared_ptr<const BidTransporterActionType::Goal> goal)
+        -> rclcpp_action::GoalResponse
+      {
+        if (this->_ongoing_transporter_bids.find(uuid) !=
+          this->_ongoing_transporter_bids.end())
+        {
+          RCLCPP_ERROR(this->get_logger(),
+            "Received duplicate bid transporter action request, rejecting...");
+          return rclcpp_action::GoalResponse::REJECT;
+        }
+        RCLCPP_INFO(this->get_logger(),
+          "Received bid transporter action request");
+        return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+      },
+      [this](const std::shared_ptr<BidTransporterGoalHandle> goal_handle)
+      -> rclcpp_action::CancelResponse
+      {
+        auto it =
+          this->_ongoing_transporter_bids.find(goal_handle->get_goal_id());
+        if (it == this->_ongoing_transporter_bids.end())
+        {
+          RCLCPP_WARN(this->get_logger(),
+            "Ongoing transporter bid not found, cancellation consider accepted "
+            "anyway...");
+          return rclcpp_action::CancelResponse::ACCEPT;
+        }
+        RCLCPP_INFO(this->get_logger(), "Bid request cancelled");
+        return rclcpp_action::CancelResponse::ACCEPT;
+      },
+      [this](const std::shared_ptr<BidTransporterGoalHandle> goal_handle)
+      {
+        this->_ongoing_transporter_bids.emplace(
+          goal_handle->get_goal_id(),
+          std::move(goal_handle));
+        RCLCPP_INFO(this->get_logger(), "Bid request added to ongoing bids");
+        this->_start_transporter_bidding(goal_handle);
```

Comment:
I have been bitten so many times in the past by `std::move` that now whenever I see one I get instantly wary. How does this actually work? Normally if you move out of a `shared_ptr` the original pointer will become null and trying to access it later _will_ segfault, as happens in this [MRP](https://godbolt.org/z/fTPffnrYs) compiler explorer snippet.